### PR TITLE
Whatsnew

### LINKED ISSUE #233: "What's new" docs section
It is standard for packages to have a "what's new" section in their documentation where they keep a record of updates with each release. We don't currently have this, but we list general features of each release on github.

### PR DESCRIPTION
**Pull request by briandleahy**
Fixes #233 , at least for the 3.3 release.

### CODE REVIEW

File: docs/source/releasenotes.rst [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+New Features
+------------
+- Inference in `holopy` has been overhauled; take a look at the updated
+  docs to check it out! Briefly, the inference and fitting modules have
+  been combined into a unified, object-oriented interface, with several
+  convenience functions available to the user both for the inference
+  strategies and the inference results. One noticeable change with this
+  is that the least-squares based fitting algorithms such as `Nmpfit`
+  can now work correctly priors, including with non-uniform priors.
```
Comment (barkls) [inline docs/source/releasenotes.rst]: "can now work correctly priors" typo

File: docs/source/releasenotes.rst [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+- A bug where `hp.propagate` failed when `gradient_filter=True` has been
+  fixed.
+- Tied parameters in inference calculations works correctly on edge
+  cases.
+- Inference should work with more generic scatterers.
+- The Fortran code should be easier to build and install on Windows
+  machines. This is partially done via a post-install script that
+  checks that files are written to the correct location (which corrects
+  some compiler differences between Windows and Linux). We still
+  recommend installing Holopy with Anaconda.
```
Comment (barkls) [inline docs/source/releasenotes.rst]: Technically just conda, not Anaconda

File: docs/source/releasenotes.rst (hunk 1 above)
Comment (briandleahy) [reply docs/source/releasenotes.rst]: Should be fixed by 2761e47

File: docs/source/releasenotes.rst (hunk 2 above)
Comment (briandleahy) [reply docs/source/releasenotes.rst]: Should be fixed by 2761e47

**Pull request merged**